src/note.rs

```diff
@@ -151,9 +160,11 @@ impl Note {
     /// Defined in [Zcash Protocol Spec § 4.8.3: Dummy Notes (Orchard)][orcharddummynotes].
     ///
     /// [orcharddummynotes]: https://zips.z.cash/protocol/nu5.pdf#orcharddummynotes
+    /// TODO zsa: remove note_type
```

Comment:
Why? I think it is necessary to be able to specify the asset when creating a dummy output note, so that it matches the asset of the note being spent in the same action. (There are no non-native dummy spent notes, because a real note with the `split_flag` set is used instead.)

Reply:
Right, my initial thinking was that we don't need `dummy_note` for zsa because of split notes. But we do need them for zsa output.